### PR TITLE
#1659 skip incomplete check for non-primary foreign key

### PR DESCRIPTION
**Pull request by chuongnh3atgmaildotcom**
Related Issue: #1659

### CODE REVIEW

File: src/Propel/Generator/Model/ForeignKey.php [hunk 1]
```diff
@@ -835,6 +835,31 @@ public function isForeignPrimaryKey()
             && !array_diff($foreignPKCols, $foreignCols));
     }
 
+    /**
+     * Returns whether this foreign key is not the primary key of the foreign
+     * table.
+     *
+     * @return bool Returns true if all columns inside this foreign key are not primary keys of the foreign table
+     */
+    public function isForeignNonPrimaryKey()
```
Comment (dereuromark) [inline src/Propel/Generator/Model/ForeignKey.php:844]: Do we want to add `: bool` typehint here?

Comment (chuongnh3atgmaildotcom) [reply src/Propel/Generator/Model/ForeignKey.php:844]: Could you please explain why it is BC? This is similar to above methods `isForeignPrimaryKey`

Comment (dereuromark) [reply src/Propel/Generator/Model/ForeignKey.php:844]: This method does not exist. So anyone that extended this class will not have a problem with the typehint here
Adding it to other existing methods will cause a fatal error on extended classes.

**Review by dereuromark — APPROVED** [review]

**Review by a-sabaa — APPROVED** [review]

File: tests/Propel/Tests/Issues/Issue1659Test.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+use Propel\Tests\TestCaseFixtures;
+
+/**
+ * Test : Propel should not allow incomplete foreign key references when foreign table has a composite primary key
+ * Issue described in https://github.com/propelorm/Propel2/issues/675.
+ * Originally described in http://stackoverflow.com/questions/7947085/are-incomplete-key-references-in-propel-useful
+ *
+ * @group model
+ */
+class Issue1659Test extends TestCaseFixtures
```
Comment (matweew) [inline tests/Propel/Tests/Issues/Issue1659Test.php:21]: We need to find better name for this class

Comment (dereuromark) [reply tests/Propel/Tests/Issues/Issue1659Test.php:21]: Sure, that can be done as follow up. For now I guess people find the details when needed.

**Pull request merged**